compiler/sem/closureiters.nim

```diff
@@ -397,20 +394,15 @@ proc collectExceptState(ctx: var Ctx, n: PNode): PNode {.inline.} =
 proc addElseToExcept(ctx: var Ctx, n: PNode) =
   if n.kind == nkStmtList and n[1].kind == nkIfStmt and n[1][^1].kind != nkElse:
     # Not all cases are covered
-    let branchBody = newNodeI(nkStmtList, n.info)
-
-    block: # :unrollFinally = true
-      branchBody.add(newTree(nkAsgn,
-        ctx.newUnrollFinallyAccess(n.info),
-        newIntTypeNode(1, ctx.g.getSysType(n.info, tyBool))))
-
-    block: # :curExc = getCurrentException()
-      branchBody.add(newTree(nkAsgn,
-        ctx.newCurExcAccess(),
-        ctx.g.callCodegenProc("getCurrentException")))
-
-    block: # goto nearestFinally
-      branchBody.add(newTree(nkGotoState, ctx.g.newIntLit(n.info, ctx.nearestFinally)))
+    let branchBody = newTreeI(nkStmtList, n.info):
+      [newTree(nkAsgn, # :unrollFinally = true
+         [ctx.newUnrollFinallyAccess(n.info),
+          newIntTypeNode(1, ctx.g.getSysType(n.info, tyBool))]),
+       newTree(nkAsgn, # :curExc = getCurrentException()
+         [ctx.newCurExcAccess(),
+          ctx.g.callCodegenProc("getCurrentException")]),
+       newTree(nkGotoState, # goto nearestFinally
+         ctx.g.newIntLit(n.info, ctx.nearestFinally))]
```

Comment:
This would make it a bit easier to read, esp when the important part is the node kind.

Also, I'm not sure about this but doesn't `newTree` take a `varargs[PNode]`? If it does then you can remove the inner `[]` as well.

```suggestion
      [nkAsgn.newTree( # :unrollFinally = true
         [ctx.newUnrollFinallyAccess(n.info),
          newIntTypeNode(1, ctx.g.getSysType(n.info, tyBool))]
       ),
       nkAsgn.newTree( # :curExc = getCurrentException()
         [ctx.newCurExcAccess(),
          ctx.g.callCodegenProc("getCurrentException")]
       ),
       nkGotoState.newTree( # goto nearestFinally
         ctx.g.newIntLit(n.info, ctx.nearestFinally)
       )]
```

Reply:
Yeah, the brackets can be removed, but I put them there to differentiate better between the info/typ arguments and the children.